src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2589,16 +2603,17 @@ private EvmExceptionType InstructionSelfDestruct<TTracing>(EvmState vmState, ref
         bool inheritorAccountExists = _state.AccountExists(inheritor);
 
         // get the verkle witness cost and charge the account access gas if the verkle cost is zero
-        var selfDestructGas = vmState.Env.Witness.AccessForSelfDestruct(executingAccount, inheritor, contractBalance.IsZero, inheritorAccountExists);
-        if (selfDestructGas == 0)
+        var gasBefore = gasAvailable;
+        var result = vmState.Env.Witness.AccessForSelfDestruct(executingAccount, inheritor, ref gasAvailable, contractBalance.IsZero, inheritorAccountExists);
+        if (gasBefore == gasAvailable)
         {
             if (!ChargeAccountAccessGas(ref gasAvailable, vmState, inheritor, spec, chargeForWarm: false,
                     opCode: Instruction.SELFDESTRUCT)) return EvmExceptionType.OutOfGas;
         }
         else
         {
             if (spec.UseHotAndColdStorage) vmState.WarmUp(inheritor);
-            if (!UpdateGas(selfDestructGas, ref gasAvailable)) return EvmExceptionType.OutOfGas;
+            if (!result) return EvmExceptionType.OutOfGas;
```

Comment:
fail as soon as you find that there is no enough gas. 
interesting that it was not doing that before the change as well.

Reply:
fixed